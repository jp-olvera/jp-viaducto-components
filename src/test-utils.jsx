import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
