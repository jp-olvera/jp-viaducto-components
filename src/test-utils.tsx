import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
