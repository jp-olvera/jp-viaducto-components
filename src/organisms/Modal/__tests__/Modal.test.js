/* eslint-env jest */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../../test-utils';
import { Modal } from '..';

describe('<Modal></Modal>', () => {
  test('should not be visible', () => {
    render(
      <Modal
        headComponent='AAA'
        onAccept={null}
        maxWidth={null}
        onReject={null}
        breakpoint={null}
        allowClickOutside={false}
      />,
    );
    expect(screen.getByTestId('overlay')).not.toBeVisible();
    expect(screen.getByTestId('modal')).not.toBeVisible();
  });
  test('should be visible', () => {
    render(<Modal active />);
    expect(screen.getByTestId('overlay')).toBeVisible();
    expect(screen.getByTestId('modal')).toBeVisible();
  });
  test('controls should be visible', () => {
    const onReject = jest.fn();
    render(<Modal active onReject={onReject} />);
    expect(screen.getByTestId('controls')).toBeVisible();
  });
  test('click close button should call handleActive function', () => {
    const handleActive = jest.fn();
    render(<Modal active handleActive={handleActive} />);
    fireEvent.click(screen.getByTestId('close-button'));
    expect(handleActive).toBeCalled();
  });
  test('click on the overlay should call handleActive function', () => {
    const handleActive = jest.fn();
    render(<Modal active handleActive={handleActive} />);
    fireEvent.click(screen.getByTestId('overlay'));
    expect(handleActive).toBeCalled();
  });
  test('should call onAccept function', () => {
    const onAccept = jest.fn();
    render(<Modal active onAccept={onAccept} />);
    fireEvent.click(screen.getByTestId('accept'));
    expect(onAccept).toBeCalled();
  });
  test('should call onReject function', () => {
    const onReject = jest.fn();
    render(<Modal active onReject={onReject} />);
    fireEvent.click(screen.getByTestId('reject'));
    expect(onReject).toBeCalled();
  });
});
