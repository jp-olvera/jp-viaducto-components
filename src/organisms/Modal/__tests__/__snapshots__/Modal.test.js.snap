// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal></Modal> should click outside of the modal component 1`] = `
<div>
  <div
    id="a"
  >
    <div
      data-testid="overlay"
      style="align-items: center; background-color: rgba(255, 255, 255, 0.5); display: none; height: 100%; justify-content: center; left: 0px; position: fixed; top: 0px; width: 100%; overflow-y: auto; z-index: 9999;"
    >
      <div
        class="StyledModal-sc-1fk6vwp-0 jbDWFx"
        data-testid="modal"
        elevation="1"
      >
        <div
          class="modal-header"
          style="flex-direction: row;"
        >
          <div>
            <h1
              class="StyledTitle-sc-1w2mjy9-0 gQMWFF"
              color="dark"
            />
          </div>
          <button
            class="BareButton__StyledButton-sc-7bpup3-0 fpYbxH"
            style="margin-left: auto;"
            type="button"
          >
            <svg
              aria-labelledby="closeIconTitle"
              data-testid="close-button"
              fill="none"
              height="1em"
              role="img"
              stroke="#9EA0A5"
              stroke-linecap="square"
              stroke-linejoin="miter"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18px"
            >
              <title
                id="closeIconTitle"
              >
                Close
              </title>
              <path
                d="M6.34314575 6.34314575L17.6568542 17.6568542M6.34314575 17.6568542L17.6568542 6.34314575"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal-content"
        />
        <div
          class="modal-bottom"
          data-testid="controls"
        >
          <button
            class="StyledButton-hwi8uc-0 dJxCps"
            data-testid="reject"
            height="2.488rem"
            type="button"
          >
            <span>
              reject
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
